src/app/services/constraintService.spec.ts

```diff
@@ -77,23 +128,86 @@ function evaluateNodeConstraintWithTwoRemovalCriteria() {
 }
 
 function evaluateCriterias() {
-  it('should evaluate criterias when it is passed one criteria that is false', () => {
-    const criterias = [criteria1];
-    spyOn(dataService, 'isCompleted').and.returnValue(false);
-    expect(service.evaluateCriterias(criterias)).toEqual(false);
+  describe('evaluateCriterias()', () => {
+    it('should evaluate criterias when it is passed one criteria that is false', () => {
+      const criterias = [criteria1];
+      spyOn(dataService, 'isCompleted').and.returnValue(false);
+      expect(service.evaluateCriterias(criterias)).toEqual(false);
+    });
+    it('should evaluate criterias when it is passed one criteria that is true', () => {
+      const criterias = [criteria1];
+      spyOn(dataService, 'isCompleted').and.returnValue(true);
+      expect(service.evaluateCriterias(criterias)).toEqual(true);
+    });
+    it('should evaluate criterias when it is passed multiple criterias and one is false', () => {
+      const criterias = [criteria1, criteria2];
+      spyOn(dataService, 'isCompleted')
+        .withArgs(nodeId1)
+        .and.returnValue(true)
+        .withArgs(nodeId2)
+        .and.returnValue(false);
+      expect(service.evaluateCriterias(criterias)).toEqual(false);
+    });
   });
-  it('should evaluate criterias when it is passed one criteria that is true', () => {
-    const criterias = [criteria1];
-    spyOn(dataService, 'isCompleted').and.returnValue(true);
-    expect(service.evaluateCriterias(criterias)).toEqual(true);
+}
+
+function getConstraintsThatAffectNode() {
+  const constraint1: Constraint = new Constraint({
+    id: 'constraint1',
+    action: 'makeAllNodesAfterThisNotVisible',
+    targetId: nodeId5
   });
-  it('should evaluate criterias when it is passed multiple criterias and one is false', () => {
-    const criterias = [criteria1, criteria2];
-    spyOn(dataService, 'isCompleted')
-      .withArgs('node1')
-      .and.returnValue(true)
-      .withArgs('node2')
-      .and.returnValue(false);
-    expect(service.evaluateCriterias(criterias)).toEqual(false);
+  const constraint2: Constraint = new Constraint({
+    id: 'constraint2',
+    action: 'makeThisNodeNotVisitable',
+    targetId: nodeId2
+  });
+  const constraint3: Constraint = new Constraint({
+    id: 'constraint3',
+    action: 'makeThisNodeNotVisitable',
+    targetId: groupId2
+  });
+  describe('getConstraintsThatAffectNode()', () => {
+    beforeEach(() => {
+      spyOn(configService, 'getConfigParam').and.returnValue(true);
+      service.activeConstraints = [constraint1, constraint2, constraint3];
+    });
+    it(`should get the constraints that affect the node when there are no constraints that affect
```

Comment:
I think we should say what we expect the function to return to make the test clearer
```suggestion
    it(`should return empty array when there are no constraints that affect
```